adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala

```diff
@@ -207,7 +210,11 @@ class Transform(protected val args: TransformArgs) extends BDGSparkCommand[Trans
       }
 
       log.info("Sorting reads")
-      adamRecords = oldRdd.sortReadsByReferencePosition()
+      if (args.sortLexicographically) {
+        adamRecords = oldRdd.sortReadsByReferencePosition()
```

Comment:
Opened #1053 for this. I agree that it needs a good cleaning, but I'd like to hold off until after 0.20.0 due to bandwidth/schedule limitations.